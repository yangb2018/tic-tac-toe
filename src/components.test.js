import { Square, Board } from './components';
import renderer from 'react-test-renderer';
import React from 'react';

test('should show square', () => {
    expect.assertions(2);
    const tree = renderer.create(<Square value=""></Square>)
        .toJSON();
    const treex = renderer.create(<Square value="ABC"></Square>)
        .toJSON();
    expect(tree).toMatchSnapshot();
    expect(treex).toMatchSnapshot();
});


it('should show Board', () => {
    const tree = renderer
      .create(<Board squares={Array(9).fill(null)}></Board>)
      .toJSON();
    
    expect(tree).toMatchSnapshot();
});