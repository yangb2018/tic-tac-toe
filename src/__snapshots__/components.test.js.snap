// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show Board 1`] = `
<div>
  <div
    className="board-row"
  >
    <button
      className="square"
      onClick={[Function]}
    />
    <button
      className="square"
      onClick={[Function]}
    />
    <button
      className="square"
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
  >
    <button
      className="square"
      onClick={[Function]}
    />
    <button
      className="square"
      onClick={[Function]}
    />
    <button
      className="square"
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
  >
    <button
      className="square"
      onClick={[Function]}
    />
    <button
      className="square"
      onClick={[Function]}
    />
    <button
      className="square"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`should show square 1`] = `
<button
  className="square"
>
  
</button>
`;

exports[`should show square 2`] = `
<button
  className="square"
>
  ABC
</button>
`;
